packages/logger-plugin/src/logger.plugin.ts

```diff
@@ -18,94 +19,22 @@ export class NgxsLoggerPlugin implements NgxsPlugin {
       return next(state, event);
     }
 
-    const options = this._options || {};
-    const logger = options.logger || console;
-    const actionName = getActionTypeFromInstance(event);
-    const time = new Date();
+    const logWriter = new LogWriter(this._options);
+    // Retrieve lazily to avoid cyclic dependency exception
+    const store = this._injector.get<Store>(Store);
```

Comment:
```ts
class NgxsLoggerPlugin {
   myMethodExample(event) {
     const actionLogger = new ActionLogger(event, this.store, this.logWriter)
   }

   private get store(): Store {
     return this._injector.get<Store>(Store);
   }

   private get logWriter(): LogWriter {
    return new LogWriter(this._options);
   }

}

```

Reply:
This introduces abstraction without adding clarity. I don't think that this is necessary.

Reply:
up to you

Reply:
Left it as is.